src/sagas/admin.js

```diff
@@ -68,33 +90,56 @@ export function* deployStore({ payload }) {
     });
     const gasPrice = (yield web3.eth.getGasPrice()) * 5;
     const gasLimit = (yield deployment.estimateGas()) * 2;
+    const transactionCostInEthers = web3.utils.fromWei(
+      (gasPrice * gasLimit).toString(),
+      "ether"
+    );
 
-    const txHash = yield sendTxWrapper({
-      txObject: deployment,
-      gasPrice,
-      gasLimit,
-      fromAddress
-    });
-
-    yield put({
-      type: types.DEPLOYING_STORE_TX_SUBMITTED,
-      payload: txHash
-    });
+    if (accountBalance >= transactionCostInEthers) {
+      let messageToast = null;
+      messageToast = toast("Deploying new certificate store...", {
```

Comment:
Can we show  "Deploying new document store..." After the transaction is signed instead of when it is sent to the signer? 

```
const deferredTx = sendTxWrapper({
        txObject: deployment,
        gasPrice,
        gasLimit,
        fromAddress
});
toast("Message");
const txHash = yield deferredTx; 
```
Might work

Reply:
Updated.